packages/eslint-plugin/src/rules/no-floating-promises.ts

```diff
@@ -323,6 +371,31 @@ export default createRule<Options, MessageId>({
   },
 });
 
+/**
+ * It checks whether a node's type matches one of the types listed in the `allowForKnownSafePromises` config
+ * @param services services variable passed from context function to check the type of a node
+ * @param node the node whose type is to be calculated to know whether it is a safe promise or not
+ * @param options The config object of `allowForKnownSafePromises`
+ * @returns `true` if the type matches, `false` if it isn't
+ */
+function doesTypeMatchesSpecifier(
```

Comment:
[Docs] A bit of naming grammar:

```suggestion
function doesTypeMatcheSpecifier(
```